File: frontend/src/app/services/general-game.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {GameBaseInformation, GameIdentifier} from "../dto/game";
import {Observable} from "rxjs";
import {environment} from "../../environments/environment";

const GAME_GENERAL_URI = environment.BASE_URI + environment.general.GAME_API + "general/";

@Injectable({
  providedIn: 'root'
})
export class GeneralGameService {

  constructor(private httpClient: HttpClient) {
  }

  public getActiveGameBaseInformation(
    gameIdentifier: GameIdentifier
  ): Observable<GameBaseInformation> {
    return this.httpClient.get<GameBaseInformation>(GAME_GENERAL_URI + "active_base_info/" + gameIdentifier.token);
  }

  public getIndividualGameBaseInformation(
    id: number
  ): Observable<GameBaseInformation> {
    return this.httpClient.get<GameBaseInformation>(GAME_GENERAL_URI + "individual_base_info/" + id);
  }

}

File: backend/src/main/java/com/fschoen/parlorplace/backend/service/game/GeneralGameService.java
package com.fschoen.parlorplace.backend.service.game;

import com.fschoen.parlorplace.backend.entity.Game;
import com.fschoen.parlorplace.backend.entity.GameIdentifier;

public interface GeneralGameService {

    Game<?, ?, ?, ?> getActiveGameBaseInformation(GameIdentifier gameIdentifier);

    Game<?, ?, ?, ?> getIndividualGameBaseInformation(Long id);

}
